### PR TITLE
Milestone 2:User interface

### PR DESCRIPTION
**Pull request by Murodjon000**
### Hi TSE
**main.rb** file responsible for user interface

### CODE REVIEW

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @Murodjon000  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] I was not able to test your project since whenever I run it, and press any key as it says on commandline, the game terminate with the following error message
Screenshot from 2021-01-05 10-42-35,

=> Make sure that the game doesn't crash when we run it

- [ ] Nice on the implemented requirements so far, but there are some remaining steps that need to be implemented:
:point_right: You should implement the full game flow, read gameflow required requirements down :point_down:

    - [ ] game tells which player turn it is (without computing who the winner is)

    - [ ] game asks the player to select from available moves

    - [ ] game informs player if the selected move is invalid (without computing what an invalid move is)

    - [ ] game displays board after the player move (without computing which spots on the board are taken)

    - [ ] game informs player if the selected move is a winning move (without computing the winning move)

    - [ ] game informs player if the selected move is a draw move (without computing draw)

    - [ ] game repeats all actions for the next player's move
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      <<-BOARD
+      1 | 2 | 3
+    ------------
+      4 | 5 | 6
+    ------------
+      7 | 8 | 9
+      BOARD
+    end
+  
+end
```
Comment (PhillipUg) [inline bin/main.rb]: - [ ] You can use all this Logic in the next milestone. In this milestone, you're only required to mock the game flow using things like a loop to imitate back and forth game play.

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey team, 👋

You have done an amazing job implementing the requirements for this project.

However, there are still some changes 👇 you need to implement to approve this request 🔍

### Changes Required ♻️ 

- [ ] No game logic in this milestone

> In this step students are supposed to only mock the basic flow of the game. This means implementing a user interface that displays prompt for users (print/puts) and collects user input (gets.chomp). No game logic (for example checking who is a winner) is required in this milestone - instead, students can just display "The winner is X" no mater who won. Usage of inline comments in this milestone is allowed. (screenshot from Pathwright)

#### Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime**
I would be happy to help! 👩‍💻 

Happy coding 🤓

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey team, 👋

You have done an amazing job implementing the requirements for this project.

However, there are still some changes 👇 you need to implement to approve this request 🔍

### Changes Required ♻️ 

- [ ] All you are remaining with is to display the board through each turn

> `[REQUIRED]` game displays board after the player move (without computing which spots on the board are taken)

#### Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime**
I would be happy to help! 👩‍💻 

Happy coding 🤓

**Comment by Murodjon000** [comment]
Hi Tse
the game displays board after the player move (without computing which spots on the board are taken)
for this comment, every time the player select number and It displays "Name selected number".Whats wrong 
thanks

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Okay what they mean by "without computing which spots on the board are taken" is for cases where another user tries to play in a spot that has already been played(taken). So you don't have to add the logic for checking whether a certain spot on the board has already been taken.

But you need to display the board after every move just like you were doing in the beginning.
So if a user selects 3, display the board and show the user letter occupying 3 on the board.

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

**Review by descholar-ceo — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
## :clap: :clap: :clap: Congratulations @Murodjon000  , your project is approved :tada: :tada: :tada: 
You have implemented all minimum requirements for this project to mark it as complete, it doesn't need more changes, you can go ahead and merge all of the changes on base branch,

Don't forget to fill the mandatory project completion form, and see you to next project,

Happy coding :computer: :computer: :computer: 

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Pull request merged**